src/sss_client/pam_sss.c

```diff
@@ -1905,14 +2166,16 @@ static int check_login_token_name(pam_handle_t *pamh, struct pam_items *pi,
     char *prompt = NULL;
     size_t size;
     char *answer = NULL;
+    /* TODO: check multiple cert case */
+    struct cert_auth_info *cai = pi->cert_list;
 
     login_token_name = getenv("PKCS11_LOGIN_TOKEN_NAME");
     if (login_token_name == NULL) {
         return PAM_SUCCESS;
     }
 
-    while (pi->token_name == NULL
-            || strcmp(login_token_name, pi->token_name) != 0) {
+    while (cai->token_name == NULL
```

Comment:
Clang says that `cai` can be NULL here

* call pam_sss
* line 2245 -> call get_pam_items
* line 1253 -> "pi->cert_list = NULL"
* return back from call get_pam_items
* line 2263 -> use case SSS_APM_AUTHENTICAte in switch
* line 2287 -> call check_login_token_name
* line 2170 -> "struct cert_auth_info *cai = pi->cert_list;"
* line 2177 -> note: Access to field 'token_name' results in a dereference of a null pointer (loaded from variable 'cai')

But I think it is false positive because we call send_and_receive-> eval_response -> parse_cert_info
What do you think?

Reply:
I added a check because we cannot be sure that the PAM responder did add the needed data.